### PR TITLE
docs: add CircleCI article

### PR DESCRIPTION
**Pull request by devcatalin**
## Pull request description 

Created article with examples on how to integrate Testkube with CircleCI.

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Comment by exu** [comment]
@jfermi worth to check :)

File: docs/docs/articles/circleci.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# Testkube CircleCI
+
+The Testkube CircleCI integration facilitates the installation of Testkube and allows the execution of any [Testkube CLI](https://docs.testkube.io/cli/testkube) command within a CircleCI pipeline. This integration can be seamlessly incorporated into your CircleCI repositories to enhance your CI/CD workflows.
+The integration offers a versatile approach to align with your pipeline requirements and is compatible with Testkube Pro, Testkube Enterprise, and the open-source Testkube platform. It enables CitcleCI users to leverage the powerful features of Testkube directly within their CI/CD pipelines, ensuring efficient and flexible test execution.
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:4]:
```suggestion
The integration offers a versatile approach to align with your pipeline requirements and is compatible with Testkube Pro, Testkube Enterprise, and the open-source Testkube platform. It enables CircleCI users to leverage the powerful features of Testkube directly within their CI/CD pipelines, ensuring efficient and flexible test execution.
```

File: docs/docs/articles/circleci.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+          name: "Trigger testkube test"
+          command: "testkube run test test-name -f"
+
+workflows:
+  run-tests-workflow:
+    jobs:
+      - run-tests
+```
+
+It is recommended that sensitive values should never be stored as plaintext in workflow files, but rather as [project variables](https://circleci.com/docs/set-environment-variable/#set-an-environment-variable-in-a-project).  Secrets can be configured at the organization or project level, and allow you to store sensitive information in CircleCI.
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:41]:
```suggestion
It is recommended that sensitive values should never be stored as plaintext in workflow files, but rather as [project variables](https://circleci.com/docs/set-environment-variable/#set-an-environment-variable-in-a-project).  Secrets can be configured at the organization or project level and allow you to store sensitive information in CircleCI.
```

File: docs/docs/articles/circleci.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+workflows:
+  run-tests-workflow:
+    jobs:
+      - run-tests
+```
+## Testkube OSS
+
+### How to configure Testkube CLI action for TK OSS and run a test
+
+To connect to the self-hosted instance, you need to have **kubectl** configured for accessing your Kubernetes cluster, and pass an optional namespace, if Testkube is not deployed in the default **testkube** namespace.
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:68]:
```suggestion
To connect to the self-hosted instance, you need to have **kubectl** configured for accessing your Kubernetes cluster and pass an optional namespace, if Testkube is not deployed in the default **testkube** namespace. 
```

File: docs/docs/articles/circleci.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+```
+## Testkube OSS
+
+### How to configure Testkube CLI action for TK OSS and run a test
+
+To connect to the self-hosted instance, you need to have **kubectl** configured for accessing your Kubernetes cluster, and pass an optional namespace, if Testkube is not deployed in the default **testkube** namespace. 
+
+If a test is already created, you can run it using the command `testkube run test test-name -f` . However, if you need to create a test in this workflow, please add a creation command, e.g.: `testkube create test --name test-name --file path_to_file.json`.
+
+In order to connect to your own cluster, you can put the your kubeconfig file into CircleCI variable named KUBECONFIGFILE
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:72]:
```suggestion
In order to connect to your own cluster, you can put your kubeconfig file into CircleCI variable named KUBECONFIGFILE.
```

File: docs/docs/articles/circleci.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+  run-tests-workflow:
+    jobs:
+      - run-tests
+```
+
+The steps to connect to your Kubernetes cluster differ for each provider. You should check the docs of your Cloud provider for how to connect to the Kubernetes cluster from CircleCI.
+
+### How to configure Testkube CLI action for TK OSS and run a test
+
+This workflow establishes a connection to EKS cluster and creates and runs a test using TK CLI. In this example we also use CircleCI variables not to reveal sensitive data. Please make sure that the following points are satisfied:
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:105]:
```suggestion
This workflow establishes a connection to the EKS cluster and creates and runs a test using TK CLI. In this example we also use CircleCI variables not to reveal sensitive data. Please make sure that the following points are satisfied:
```

File: docs/docs/articles/circleci.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+- The **_AwsAccessKeyId_**, **_AwsSecretAccessKeyId_** secrets should contain your AWS IAM keys with proper permissions to connect to EKS cluster.
+- The **_AwsRegion_** secret should contain AWS region where EKS is
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:107]:
```suggestion
- The **_AwsRegion_** secret should contain the AWS region where EKS is.
```

File: docs/docs/articles/circleci.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+- Tke **EksClusterName** secret points to the name of EKS cluster you want to connect.
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:108]:
```suggestion
- Tke **EksClusterName** secret points to the name of the EKS cluster you want to connect.
```

File: docs/docs/articles/circleci.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+    jobs:
+      - setup-aws
+      - run-testkube-on-aws:
+          requires:
+            - setup-aws
+```
+
+### How to connect to GKE (Google Kubernetes Engine) cluster and run a test 
+
+This example connects to a k8s cluster in Google Cloud, creates and runs a test using Testkube CircleCI. Please make sure that the following points are satisfied:
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:153]:
```suggestion
This example connects to a k8s cluster in Google Cloud then creates and runs a test using Testkube CircleCI. Please make sure that the following points are satisfied:
```

File: docs/docs/articles/circleci.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+- The **_GKE Sevice Account_** should be created prior in Google Cloud and added to CircleCI variables along with **_GKE Project_** value;
```
Comment (jfermi) [inline docs/docs/articles/circleci.md:154]:
```suggestion
- The **_GKE Sevice Account_** should already be created in Google Cloud and added to CircleCI variables along with **_GKE Project_** value.
```

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

**Review by nicufk — APPROVED** [review]

**Pull request merged**